cmd/provisioner-localpv/app/provisioner_hostpath.go

```diff
@@ -162,28 +169,29 @@ func (p *Provisioner) DeleteHostPath(pv *v1.PersistentVolume) (err error) {
 		return errors.Errorf("no HostPath set")
 	}
 
-	hostname := pvObj.GetAffinitedNodeHostname()
-	if hostname == "" {
-		return errors.Errorf("cannot find affinited node hostname")
+	nodeAffinityKey, nodeAffinityValue := pvObj.GetAffinitedNodeLabelKeyAndValue()
+	if nodeAffinityValue == "" {
+		return errors.Errorf("cannot find affinited node details")
 	}
-	alertlog.Logger.Infof("Get the Node Object from hostName: %v", hostname)
+	alertlog.Logger.Infof("Get the Node Object with label %v : %v", nodeAffinityKey, nodeAffinityValue)
 
 	//Get the node Object once again to get updated Taints.
-	nodeObject, err := p.GetNodeObjectFromHostName(hostname)
+	nodeObject, err := p.GetNodeObjectFromLabels(nodeAffinityKey, nodeAffinityValue)
```

Comment:
Can there be multiple nodes with same labels, for this should we use hostname?

Reply:
good one. There be a corner case where old node and new node may co-exist. Added a check to return error in such cases. 

The expectation is that the custom node affinity label is always unique.